specs/agents/tracing-instrumentation-messaging.md

```diff
@@ -0,0 +1,103 @@
+
+For RabbitMQ transaction and polling spans, the queue name is used instead, whenever available (i.e. when the polling yields a message).
+
+### Context fields
+
+- **`context.message.queue.name`**: optional for `messaging` spans and transactions. Indexed as keyword. Wherever the broker terminology 
+uses "topic", this field will contain the topic name. In RabbitMQ, whenever the queue name is not available, use exchange name instead.
+- **`context.message.age.ms`**: optional for message/record receivers only (transactions or spans). 
+A numeric field indicating the message's age in milliseconds. Relevant for transactions and 
+`receive` spans that receive valid messages. There is no accurate definition as to how this is calculated. If the messaging framework 
+provides a timestamp for the message- agents may use it. Otherwise, the sending agent can add a timestamp _indicated as milliseconds since 
+epoch UTC_ to the message's metadata to be retrieved by the receiving agent. If a timestamp is not available- agents should omit this field. 
+Clock skews between agents are ignored, unless the calculated age (receive-timestamp minus send-timestamp) is negative, in which case the 
+agent should report 0 for this field.
+- **`context.message.routing-key`**: optional. Use only where relevant. Currently only RabbitMQ.
+
+#### Transaction context fields
+
+- **`context.message.body`**: similar to HTTP requests' `context.request.body`- only fill in messaging-related **transactions** (ie 
+incoming messages creating a transaction) and not for outgoing messaging spans. 
+   - Capture only when `ELASTIC_APM_CAPTURE_BODY` config is set to `true`.
+   - Only capture UTF-8 encoded message bodies.
+   - Limit size to 10000 characters. If longer than this size, trim to 9999 and append with ellipsis
+- **`context.message.headers`**: similar to HTTP requests' `context.request.headers`- only fill in messaging-related **transactions**.
+   - Capture only when `ELASTIC_APM_CAPTURE_HEADERS` config is set to `true`.
+   - Sanitize headers with keys configured through `ELASTIC_APM_SANITIZE_FIELD_NAMES`.
+   - Intake: key-value pairs, same like `context.request.headers`.
+
+#### Span context fields
+
+- **`context.destination.address`**: optional. Not available in some cases. Only set if the actual connection is available.
+- **`context.destination.port`**: optional. Not available in some cases. Only set if the actual connection is available.
+- **`context.destination.service.name`**: mandatory. Use the framework name in lowercase, e.g. `kafka`, `rabbitmq`.
+- **`context.destination.service.resource`**: mandatory. Use the framework name in lowercase. Wherever the queue/topic/exchange name is
+ available, append it with a leading `/`, for example: `kafka/myTopic`, `rabbitmq/myExchange`, `rabbitmq`.
+- **`context.destination.service.type`**: mandatory. Use `messaging`.
+
+### `ELASTIC_APM_IGNORE_MESSAGE_QUEUES` configuration
```

Comment:
Refer to config options in their central config form e.g. `ignore_message_queues` to be consistent with other parts of the spec.